### PR TITLE
Py2 and py3 support

### PR DESCRIPTION
**Pull request by gbaty**
Support both Python 2.6+ and Python 3.4+

Main changes from python 2 to python 3 are:
'raise Exception "message"'  ->  'raise Exception("message")'
print "message"  -> print("message")
long, int  ->  int
raw_input  -> input
cPickle  ->  pickle

input  -> eval(input()) # not used in SU2, just for information

This commit also removes unused import, especially import of os, shutil and copy

Useful links:
- https://docs.python.org/3/howto/pyporting.html
- http://python3porting.com
- https://docs.python.org/3/whatsnew/3.0.html#library-changes, 4th item.

### CODE REVIEW

**Comment by gbaty** [comment]
This branch has been rebased on last develop version (commit a24f7c2f334f771f173f93ea24f9d73daec1df21, Date: Tue May 31 17:33:41 2016 -0700)

Only a few tests fail due to numerical differences but not on python syntax errors. So I think failures are due to parts not linked to python scripts that is why I keep this pull-request open and ready to merge.
Can someone confirm that ?
Thank you

**Comment by economon** [comment]
Thank you for rebasing.. indeed, we're in the middle of cleaning up a few things in the regressions at the moment, but I hope we can pull in your Python 3 contributions very soon. We may kick off the tests again here once we get things sorted.

Thanks for your patience,
Tom

**Comment by economon** [comment]
Hi,

I would like to test your improvements on a couple of different clusters. Could you please rebase once more on the latest commit in the develop branch? That way, we can run the regressions again to make sure everything is ok before merging.

Any other volunteers in the community willing to run this request on clusters of their own to help verify that the Python framework functions well on a variety of platforms?

Thanks!
Tom

**Comment by gbaty** [comment]
No problem.
I rebased my test branch on last develop version to run a first pass of test (see [travis page](https://travis-ci.org/gbaty/SU2/builds/135385229)). If there are no syntax error, I will push this rebased version here to allow you to continue tests as you suggest.

**Comment by hlkline** [comment]
Thanks for rebases
A couple of notes: 
- the reference to the github issue in the comments may be confusing to others as to which part of that thread is being referenced. To make sure that future developers are aware of import issues, maybe just one line like 'recommended import format compatible with both Python 2.7 and 3', if you feel a comment really is necessary there.
- on the regressions: I know it may seem like small differences, but the fact that they are different with just changes to the python code is very concerning, and in some of the tests the differences are quite large (specifically discadj_euler_py and directidff_euler_py). If they are calling the exact same C++ executables and performing the exact same computations the result should be the same. If this is something going on behind the scenes within the python packages that cannot be avoided, we need to understand what that difference is, and where it is coming from. 
The only explanation I can think of off the top of my head for why the output could be different with only changing python code, is if some function previously imported from a different package is using single-precision instead of double-precision for some arithmetic operation or file io. If what you changed is actually providing GREATER precision that's fantastic, but we need to know for sure either way.

File: SU2_PY/SU2/mesh/tools.py [hunk 1]
```diff
@@ -29,6 +29,11 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with SU2. If not, see <http://www.gnu.org/licenses/>.
 
+import sys
+if sys.version_info.major > 2:
+    # In PY3, 'long' and 'int' are unified in 'int' type
+    long = int
```
Comment (gbaty) [inline SU2_PY/SU2/mesh/tools.py]: In python 3, long is renamed int and is equivalent to py2 "long".
Here, line long = int is applied only in case of python 3 so normally it changes nothing on python2

**Comment by gbaty** [comment]
Hi. I think the best is just to remove reference to the issue. I think comment is explicit enough. I will commit all minor changes in next commit.

I am not sure to understand well the problem about regression but I think it was just due to tests regression failing on the specific develop commit I used as base of my branch. 

Nevertheless, there is this change that can be important for futur python changes:
https://www.python.org/dev/peps/pep-0237. I have applied this one time in my commit (see comment on pull-request diff). 

My test branch rebased on last develop commit pass all tests so normally travis should confirm that all tests are ok now for this pull request. See you soon if not ;)

**Comment by economon** [comment]
Bumping this PR. Has anyone had a chance to try this out on some clusters? We also have some conflicts now that we've waited...

**Comment by gbaty** [comment]
I can fix conflicts by rebasing this branch again but I prefer to do it when someone is ready to review it.
Please let me know.

File: SU2_PY/SU2/io/config.py [hunk 2]
```diff
@@ -278,7 +275,7 @@ def __repr__(self):
     
     def __str__(self):
         output = 'Config: %s' % self._filename
-        for k,v in self.iteritems():
+        for k,v in self.items():
```
Comment (petebachant) [inline SU2_PY/SU2/io/config.py]: Missing a space between `k` and `v`.

**Comment by fpalacios** [comment]
Hi @gbaty,
Thanks a lot for this contribution, I think it could be very important for the SU2 community and I would love to review it. 

Would you be so kind to update your branch with the latest develop branch to fix the conflicts?

Thanks again for improving SU2.

Best,
Francisco

**Comment by gbaty** [comment]
I am ok to rebase and fix conflict but I won't have much time to do it before the end of august.

**Comment by petebachant** [comment]
@gbaty I just took a stab at rebasing your branch on top of `develop`. Give it a quick check to make sure I didn't miss any conflicts: https://github.com/gbaty/SU2/pull/1

**Comment by gbaty** [comment]
Thank you very much @petebachant . Again I rebased this result (your pull request merged) on last develop version to be up to date and have a cleaner git history tree. The result is here : [gbaty/SU2/py2_and_py3_support_rebased_on_develop_20170821](https://github.com/gbaty/SU2/tree/py2_and_py3_support_rebased_on_develop_20170821)

We now need to double check all changes. I am afraid to have lost a small part of your work that is why I have created a new branch. All you work is in py2_and_py3_support.

Doing a pr on develop allow to see easily all py2 to py3 changes.

**This branch is not tested** and done quickly. Do trust it directly, I need to continue to work on it.

**Comment by petebachant** [comment]
Hi @gbaty,

I started pushing this a bit further and opened up a new PR (#424). Next step is trying to get the wrapper to build against a version of Python 3 and test with that on Travis. Let me know if you have any ideas. I'm struggling a bit with the Makefile generation.

**Comment by gbaty** [comment]
Ok! The workflow about this branch is now quite hard to follow. 
So to resume :
  - **#424 is the branch to work on**. #424 is "py2_and_py3_support" merged on a recent "develop" branch (2017/08/11)
  - #260 is obsolete and replaced by previous one. That is why I close it
